### PR TITLE
Test with CircleCI 2.0

### PR DESCRIPTION
**Pull request by etpinard**
CircleCI has mostly stopped adding new images compatible with its 1.0 framework, so it's about time we pay back some tech-debt :pick: :moneybag:  In addition, upgrading to 2.0, makes our tests run in about 10 minutes - roughly 5 minutes less than for the current master. :tada:

Note that our jasmine tests now run on Chrome 59 (no more Chrome 54 as on the current master). We're using node 6.x + npm 3.x images (same as the current master). An upcoming PR (cc https://github.com/plotly/plotly.js/issues/1798) will bump those to node 8.x + npm 5.x - which should speed up our tests even more :racing_car: 

The only _tricky_ part here was to make the image tests work. CircleCI 2.0 is setup to fetch a single docker container and run tests directly in that container. In our CircleCI 1.0 config, we fetched our `imagetest` docker container and made requests from a "main" container to a port mapped to the nw.js server inside `imagetest`. Circle 2.0 doesn't allow communication between containers, so I had to update some of our `tasks/` script, mostly by removing a few hacky lines  :stuck_out_tongue:  

About our increasingly-annoying intermittent test failures, I still noticed some unfortunately. But hopefully this is a step in the right direction.

cc @alexcjohnson @dfcreative @scjody (who's been asking me to do this for a while)

### CODE REVIEW

File: tasks/noci_test.sh [hunk 1]
```diff
@@ -10,7 +10,4 @@ npm run test-jasmine -- --tags=noCI --nowatch || EXIT_STATE=$?
 # mapbox image tests take too much resources on CI
 npm run test-image -- mapbox_* --queue || EXIT_STATE=$?
 
-# run gl2d image test again (some mocks are skipped on CI)
-npm run test-image-gl2d || EXIT_STATE=$?
```
Comment (etpinard) [inline tasks/noci_test.sh]: fixup for https://github.com/plotly/plotly.js/pull/2258/commits/9f304e658bfd8cecd6a5a011faa46e6161deaa58 

No need for this line anymore as now all gl2d mocks are tested on CI.

File: package.json [hunk 2]
```diff
@@ -44,7 +44,7 @@
     "start-image_viewer": "node devtools/image_viewer/server.js",
     "start": "npm run start-test_dashboard",
     "baseline": "node tasks/baseline.js",
-    "preversion": "npm-link-check && npm dedupe",
+    "preversion": "npm-link-check && npm dedupe && npm ls --prod",
```
Comment (etpinard) [inline package.json]: cc @alexcjohnson @dfcreative 

This commit partly reverts https://github.com/plotly/plotly.js/commit/1d3a3f0ead64168f1764ad7fe215936964bd116d

`npm ls --prod` will fail only if a bundle dependency (i.e. ignoring dev-dependencies) is (1) missing, (2) needs to be updated or (3) is extraneous. This step ensures that we're not bundling `dist/` bundles with incorrect dependencies.

File: .circleci/test.sh [hunk 3]
```diff
@@ -1,21 +1,29 @@
 #!/bin/bash
 
+# override CircleCi's default run settings
+set +e
+set +o pipefail
```
Comment (etpinard) [inline .circleci/test.sh]: I don't like CircleCI's new defaults where a given job stop executing after the first test failure. I prefer running all the tests no matter what, to see if a given commit makes multiple tests fail. I hope you agree with me.

**Comment by alexcjohnson** [comment]
LGTM! Only thing that occurs to me is we could balance the containers a bit more, perhaps taking the bundle tests out of jasmine and lumping with syntax, that would save another minute or two. But not a big deal, this is already a big win! 💃

**Comment by etpinard** [comment]
Great. Merging. Next stop updating to node 8.x. I'll should have time to do that by week's end.

**Pull request merged**